mbuild/conversion.py

```diff
@@ -0,0 +1,1388 @@
+            if coords_only:
+                tmp = read_xyz(filename)
+                if tmp.n_particles != compound.n_particles:
+                    raise ValueError('Number of atoms in {filename}'
+                        'does not match {compound}'.format(**locals()))
+                ref_and_compound = zip(tmp._particles(include_ports=False),
+                                       compound.particles(include_ports=False))
+                for ref_particle, particle in ref_and_compound:
+                    particle.pos = ref_particle.pos
+            else:
+                compound = read_xyz(filename, compound=compound)
+        elif extension == '.xyz' and 'top' in kwargs:
+            backend = 'mdtraj'
+
+    # Then pybel reader
+    elif backend == 'pybel':
+        pybel = import_('pybel')
+        if extension == '.sdf':
+            pybel_mol = pybel.readfile('sdf', filename)
+            # pybel returns a generator, so we grab the first molecule of a
+            # list of len 1.
+            # Raise ValueError if there are more molecules
+            pybel_mol = [i for i in pybel_mol]
+            if len(pybel_mol) == 1:
+                compound = from_pybel(
+                            pybel_mol=pybel_mol[0],
+                            compound=compound,
+                            coords_only=coords_only,
+                            infer_hierarchy=infer_hierarchy)
+            else:
+                raise ValueError('More tahn one pybel molecule in file,'
+                            'more than one pybel molecule is not supported')
+        # text file detected, asssume contain smiles string
+        elif extension == '.txt':
+            # Fail-safe measure
+            compound = load_smiles(filename, compound)
+
+    # Then parmed reader
+    elif backend == 'parmed':
+        warn('Using parmed reader. Bonds may be inferred '
+             'from inter-particle distances and standard '
+             'residue templates')
+        structure = pmd.load_file(filename, structure=True, **kwargs)
+        compound = from_parmed(
+                            structure=structure,
+                            compound=compound,
+                            coords_only=coords_only,
+                            infer_hierarchy=infer_hierarchy)
+
+    # Then mdtraj reader
+    elif backend == 'mdtraj':
+        traj = md.load(filename, **kwargs)
+        compound = from_trajectory(
+                            traj=traj,
+                            compound=compound,
+                            frame=-1,
+                            coords_only=coords_only,
+                            infer_hierarchy=infer_hierarchy)
+
+    if rigid:
+        compound.label_rigid_bodies()
+
+    # Note: 'Input not supported' error will be handled
+    # by the corresponding backend
+    return compound
+
+def from_mbuild(cmpd, compound=None, coords_only=False,
+                infer_hierarchy=False):
```

Comment:
Can you specify why this is needed as a separate method?

Reply:
I only have this as a separate method to be kind of 'parallel' to the other (so the other `from_*` methods). The purpose is to be able to throw the load object into the same loop (`for type_ in type_dict:` in line 137) without having to do a separate check for mb.Compound type)

Reply:
It seems like you could just have 
```python3
if isinstance(object, mb.Compound):
    return object
```
which would be a lot simpler and 30 lines of code shorter.

Reply:
This method also supposes to cover the case when there is a host compound, in which case the object will be added as a child to the host compound. but i get your point, I guess I will just remove this method.